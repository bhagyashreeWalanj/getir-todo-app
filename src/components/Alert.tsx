import React, { useEffect } from 'react'
import styles from '../styles/alert.module.scss'
import { default as classnames } from 'classnames'
import { CgDanger } from 'react-icons/cg'
import { TiTick } from 'react-icons/ti'

export interface IToDoAlert {
  type: string
  msg: string
  removeAlert: () => void
}

const Alert = ({ type, msg, removeAlert }: IToDoAlert) => {
  useEffect(() => {
    const timeout = setTimeout(() => {
      removeAlert()
    }, 3000)
    return () => clearTimeout(timeout)
  })

  return (
    <div
      className={classnames(
        type === 'success-edit'
          ? [styles.editSuccessAlert]
          : type === 'danger-edit'
          ? [styles.editErrorAlert]
          : [styles.alert],
      )}
    >
      <p
        className={classnames({
          [styles.errormsg]: type === 'danger',
          [styles.successmsg]: type === 'success',
          [styles.errormsgEdit]: type === 'danger-edit',
          [styles.successmsgEdit]: type === 'success-edit',
        })}
        id={`alertDiv`}
      >
        {type === 'danger' ? (
          <CgDanger className={styles.dangerIcon} />
        ) : (
          <TiTick />
        )}{' '}
        {msg}
      </p>
    </div>
  )
}

export default Alert
