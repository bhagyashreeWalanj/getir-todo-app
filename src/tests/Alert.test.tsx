import React from 'react'
import { shallow, mount } from 'enzyme'
import Alert from '../components/Alert'

describe('Testing Alert component', () => {
  const removeAlert = () => jest.fn()
  const msg = 'SUCCESS : DONE'
  const type = 'success'

  const wrapper = shallow(
    <Alert type={type} msg={msg} removeAlert={removeAlert} list={[]} />,
  )

  it('should display to Alert Component', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should check success class exists', () => {
    expect(wrapper.find('.successmsg')).toHaveLength(1)
  })

  it('should display message on div element', () => {
    expect(wrapper.find('.successmsg').text()).toContain('SUCCESS : DONE')
  })

  it('should check success class for edit exists', () => {
    const wrapperError = mount(
      <Alert
        type={'success-edit'}
        msg={'SUCCESS : DONE'}
        removeAlert={removeAlert}
        list={[]}
      />,
    )

    expect(wrapperError.find('.successmsgEdit')).toHaveLength(1)
    expect(wrapperError.find('.successmsgEdit').text()).toContain('SUCCESS')
  })

  it('should check error class exists', () => {
    const wrapperError = mount(
      <Alert
        type={'danger'}
        msg={'ERROR : EXCEPTION'}
        removeAlert={removeAlert}
        list={[]}
      />,
    )

    expect(wrapperError.find('.errormsg')).toHaveLength(1)
    expect(wrapperError.find('.errormsg').text()).toContain('ERROR')
  })

  it('should check error class for edit exists', () => {
    const wrapperError = mount(
      <Alert
        type={'danger-edit'}
        msg={'ERROR : EXCEPTION'}
        removeAlert={removeAlert}
        list={[]}
      />,
    )

    expect(wrapperError.find('.errormsgEdit')).toHaveLength(1)
    expect(wrapperError.find('.errormsgEdit').text()).toContain('ERROR')
  })

  it('should call remove alert Div after 3 seconds of timeout', () => {
    expect(wrapper.find('.successmsg')).toHaveLength(1)
    setTimeout(() => {
      expect(removeAlert).toBeCalledTimes(1)
    }, 3000)
  })
})
