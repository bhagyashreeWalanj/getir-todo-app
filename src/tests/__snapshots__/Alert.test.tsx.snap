// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Alert component should display to Alert Component 1`] = `
<div
  className="alert"
>
  <p
    className="successmsg"
    id="alertDiv"
  >
    <TiTick />
     
    SUCCESS : DONE
  </p>
</div>
`;
